src/main/app/case/case-api.ts

```diff
@@ -50,8 +50,8 @@ export class CaseApi {
     return this.apiClient.sendEvent(caseId, toApiFormat(userData), eventName);
   }
 
-  public async addPayment(caseId: string, payments: ListValue<Payment>[]): Promise<CaseWithId> {
-    return this.apiClient.sendEvent(caseId, { applicationPayments: payments }, CITIZEN_ADD_PAYMENT);
+  public async addPayment(caseId: string, payments: ListValue<Payment>[], eventName: string): Promise<CaseWithId> {
```

Comment:
maybe rename the function so that it's clear that one call adds the payment and the other updates